### PR TITLE
Add `frames_before` and `frames_after` to `RemoveArtifactsRecording`

### PR DESCRIPTION
**Pull request by khl02007**
@alejoe91 
#619 
How to pass in these params in 'median' and 'average' modes?

### CODE REVIEW

**Comment by alejoe91** [comment]
Thanks @khl02007 

Probably it will be a good idea to propagate this option to the extract waveforms. I suggest we add it to #1152 and wait for it to be merged before adding it here. Sounds good?

**Comment by khl02007** [comment]
@alejoe91 yeah sounds good

File: spikeinterface/preprocessing/remove_artifacts.py [hunk 1]
```diff
@@ -98,22 +116,52 @@ def __init__(self, recording, list_triggers, ms_before=0.5, ms_after=3.0, mode='
...
+            if (ms_before is not None) or (frames_before is not None):
+                raise ValueError("You must specify either `ms_after` and `frames_after`",
+                                 " (unless you want to remove single samples, in which case",
+                                 " set `ms_before` and `frames_before` to None)")
+        elif (ms_after is not None) and (frames_after is None):
             ms_after = float(ms_after)
+        elif (ms_after is None) and (frames_after is not None):
+            frames_after = int(frames_after)
+        else:
+            raise ValueError("You cannot specify both `ms_after` and `frames_after`; choose one.")
```
Comment (alejoe91) [inline spikeinterface/preprocessing/remove_artifacts.py:143]: @khl02007 IMO the logic could be simpler. If `frames_before is not None`, then assert that `frames_after` is not None and use it. By default, they will be None, so if they are used they can be prioritized over the ms_before/after.

**Comment by alejoe91** [comment]
@khl02007 after more thinking and discussion we thought that the problem of uneven sampling is not only here, but also, for example, whee extracting waveforms. 

Internally, even though we allow to `set_times()`, SpikeInterface always assumes that buffers are uniformly sampled. In case they are not, some steps of the processing, e.g., filtering, would be wrong.

Therefore, it is fair to leave `ms_before` and `ms_after`, which are only used to estimate the padding assuming samples are uniform (just dividine by sampling rate). These are in our opinion better arguments because they are more intuitive and robust agains sampling rate.

So basically, we would keep the arguments as they are, but extend the docstring with this additional piece of information.

Do you agree?

(Sorry for the dealy in replying to this!!!)

**Comment by khl02007** [comment]
@alejoe91 I suppose this means spikeinterface as a rule won't support any processing of recordings with uneven sampling. Is that right? I agree that this is a challenge. What do you suggest that we do in general? Interpolate the 'holes' before any processing steps?

**Comment by alejoe91** [comment]
Hi @khl02007 

Indeed several processing steps assume uniform sampling. In case of small gaps (say a few tens of samples), I don't think this is a problem. In case there are larger gaps, the correct way to handle it is to split the recording into multiple pseudo-continuous segments. That way each continuous block would be processed correctly. This approach should also work for the `remove_artifacts` function!

**Comment by khl02007** [comment]
@alejoe91 yes OK to close
